cmd/crank/main.go

```diff
@@ -18,20 +18,34 @@ package main
 
 import (
 	"github.com/alecthomas/kong"
-
-	"github.com/crossplane/crossplane/cmd/crank/configuration"
-	"github.com/crossplane/crossplane/cmd/crank/provider"
 )
 
+var _ = kong.Must(&cli)
+
 var cli struct {
-	Configuration configuration.Cmd `cmd:"" help:"Interact with Configurations."`
-	Provider      provider.Cmd      `cmd:"" help:"Interact with Providers."`
+	Build buildCmd `cmd:"" help:"Build Crossplane packages."`
+	Push  pushCmd  `cmd:"" help:"Push Crossplane packages."`
+}
+
+// childArg is used to pass child arguments to parent commands.
+type childArg struct {
+	strVal string
+}
+
+// strChild is a string value child argument.
+type strChild string
+
+func (c strChild) AfterApply(arg *childArg) error { // nolint:unparam
+	arg.strVal = string(c)
+	return nil
 }
```

Comment:
Any time an arg or flag of type `strChild` is traversed in the node tree, `AfterApply` will be invoked. I am using this to propagate values from child to parent commands.